### PR TITLE
TDL-19329: Update Pagination to cursor based for certain streams

### PR DESCRIPTION
**Pull request by hpatel41**
# Description of change
TDL-19329: Update Pagination to cursor-based for certain streams
- Updated the pagination to cursor-based as per the Recharge API [documentation](https://developer.rechargepayments.com/2021-11/cursor_pagination)

# Manual QA steps
 - 
 
# Risks
 - 
 
# Rollback steps
 - revert this branch

### CODE REVIEW

File: tap_recharge/streams.py [hunk 1]
```diff
@@ -247,12 +247,13 @@ def get_records(
         url = None
 
         while paging:
-            records, links = self.client.get(path, url=url, params=self.params)
+            records, _ = self.client.get(path, url=url, params=self.params)
```
Comment (savan-chovatiya) [inline tap_recharge/streams.py:250]: The tap is not using a second return value of the `client.get()` function anywhere so remove it from `get()` function itself.

File: tests/unittests/test_cursor_pagination.py [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+from tap_recharge.client import RechargeClient
+from tap_recharge.streams import Addresses
+
+COUNTER = 0
+
+def get(*args, **kwargs):
+    """
+        Function to return API response with 'next_cursor' for 2 responses and last response without 'next_cursor'
+    """
+    global COUNTER
```
Comment (dbshah1212) [inline tests/unittests/test_cursor_pagination.py:12]: Add a reason for using a global variable in the comment.

File: tap_recharge/streams.py (hunk 1 above)
Comment (harshpatel4crest) [reply tap_recharge/streams.py]: Updated the **request** function to return in json response.

File: tests/unittests/test_cursor_pagination.py (hunk 2 above)
Comment (harshpatel4crest) [reply tests/unittests/test_cursor_pagination.py]: Updated the code to not use a global variable.

**Review by savan-chovatiya — APPROVED** [review]

**Review by dbshah1212 — APPROVED** [review]

File: tests/unittests/test_request_timeout.py [hunk 3]
```diff
@@ -99,7 +99,7 @@ def test_config_provided_request_timeout(self, mock_get, mock_request):
         client = RechargeClient(**config)
         client.request("GET", "dummy_path")
         
-        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=100.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'User-Agent': 'dummy_ua'})
+        mock_request.assert_called_with('GET', 'https://api.rechargeapps.com/dummy_path', stream=True, timeout=100.0, headers={'X-Recharge-Access-Token': 'dummy_at', 'Accept': 'application/json', 'X-Recharge-Version': '2021-11', 'User-Agent': 'dummy_ua'})
```
Comment (RushiT0122) [inline tests/unittests/test_request_timeout.py:102]: Handle changes requested in PR #32 here as well.

**Review by RushiT0122 — CHANGES_REQUESTED** [review]
Need to resolve conflicts in test_request_timeout.py with PR #32.

**Comment by harshpatel4crest** [comment]
> Need to resolve conflicts in test_request_timeout.py with PR #32.

This will be covered as part of PR #32

**Review by RushiT0122 — APPROVED** [review]

**Review by RushiT0122 — APPROVED** [review]

**Review by RushiT0122 — APPROVED** [review]

**Review by KrisPersonal — APPROVED** [review]

**Pull request merged**